demo/dlb-init.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+enable_and_configure_vfs() {
+  devpath=$1
+  num_vfs_per_pf=$2
+  # enable sriov
+  sriov_numvfs_path="$devpath"/sriov_numvfs
+  if test -w "$sriov_numvfs_path"; then
+    if [ "$(cat "$sriov_numvfs_path")" -eq 0 ]; then
+      echo -n "$num_vfs_per_pf" > "$sriov_numvfs_path"
+      # configure vfs
+      for virtfnN in $(basename -a "$devpath"/virtfn*) ; do
```

Comment:
I doubt we can configure more than one VF with default values. If this is a valid doubt, do we need this loop?

Reply:
> One thing stupid in the script is.. that
> 
>     1. We can set only one vf because there is "default" values for resources (num_atomic_inflights, num_dir_credits, etc).
> 
>     2. There is still 'for' loop that is meant for setting multiple vfs...

Yeah, it is what I was questioning before. It is somewhat meaningless for now.
Or, we need to make the script really working to configure multiple vfs.

What do we do?

Reply:
What if we do like this?
1. If num_vfs_per_pf = 1, then configure the resources with default values like 2048, something as now.
2. If num_vfs_per_pf > 1, then do as we tried to do before (total_resources ÷ num_vfs_per_vf).

Reply:
Or, do we just have it in the future, and for now, just only 1 vf?

Reply:
> If num_vfs_per_pf = 1, then configure the resources with default values like 2048, something as now.
> If num_vfs_per_pf > 1, then do as we tried to do before (total_resources ÷ num_vfs_per_vf).

I don't think we need this. hint: total_resources ÷ 1 == total_resources

Reply:
I mean that if we want to make the num_vfs_per_pf "configurable" as we planned before, then num_vfs_per_pf can be more than 1. Then, it should be divided like that.
For now, it seems that we just go without configurable variable. let me just remove all things related to that and set only virtfn0 and vf0_resource.

Reply:
It's a little bit stupid because the end result for now is just... exactly same as copying the readme file of dlb plugin into a script.... Idk why we had to waste so much time...

Reply:
> I mean that if we want to make the num_vfs_per_pf "configurable" as we planned before, then num_vfs_per_pf can be more than 1. Then, it should be divided like that.

ok, let me explain what I've said above with more details:
when num_vfs_per_pf = 1 we can use echo -n 2048 / 1 > "$vf_resources_path/num_atomic_inflights" and it will be correct.
when num_vfs_per_pf = 2 (or any other positive number) we still can use echo -n 2048 / num_vfs_per_pf > "$vf_resources_path/num_atomic_inflights" and it will be correct

So, we don't need to distinguish between those 2 cases as you described. Is it more clear now?

Reply:
> It's a little bit stupid because the end result for now is just... exactly same as copying the readme file of dlb plugin into a script.... Idk why we had to waste so much time...

Not sure I follow. Can you elaborate?

Reply:
> > I mean that if we want to make the num_vfs_per_pf "configurable" as we planned before, then num_vfs_per_pf can be more than 1. Then, it should be divided like that.
> 
> ok, let me explain what I've said above with more details: when num_vfs_per_pf = 1 we can use echo -n 2048 / 1 > "$vf_resources_path/num_atomic_inflights" and it will be correct. when num_vfs_per_pf = 2 (or any other positive number) we still can use echo -n 2048 / num_vfs_per_pf > "$vf_resources_path/num_atomic_inflights" and it will be correct
> 
> So, we don't need to distinguish between those 2 cases as you described. Is it more clear now?

@bart0sh
Of course, I know 2048 / 1 = 2048. And,, it used to be always like as you said. I think you did not understand what I meant..!

What I meant is not 2048 / 1, but the 'recommended' values. As I told many times already, 2048 is not same as "total_resource". It is just a recommended value.
Let me give also more detailed examples.

But, my suggestion was
```bash
if num_vfs_per_pf = 1
  echo -n 2048 > "$vf_resources_path/num_atomic_inflights"
  echo -n 2048 > "$vf_resources_path/num_dir_credits"
  echo -n 8 > "$vf_resources_path/num_dir_ports"
  echo -n 2048 > "$vf_resources_path/num_hist_list_entries"
  echo -n 8192 > "$vf_resources_path/num_ldb_credits"
  echo -n 4 > "$vf_resources_path/num_ldb_ports"
  echo -n 32 > "$vf_resources_path/num_ldb_queues"
  echo -n 32 > "$vf_resources_path/num_sched_domains"
  echo -n 2 > "$vf_resources_path/num_sn0_slots"
  echo -n 2 > "$vf_resources_path/num_sn1_slots"
if num_vfs_per_pf > 1
  echo -n total_resources/num_atomic_inflights / num_vfs_per_pf > "$vf_resources_path/num_atomic_inflights"
  echo -n total_resources/num_dir_credits / num_vfs_per_pf > "$vf_resources_path/num_dir_credits"
  echo -n total_resources/num_dir_ports / num_vfs_per_pf > "$vf_resources_path/num_dir_ports"
...
```

Here what you need to consider is that the recommended values for the case num_vfs_per_pf =1 is not same as total_resources/<resource_name>.

IMO, it is meaingless now to discuss this matter since we agreed to have just one vf and there is no option for the number of vfs per pf, right?

Reply:
> Here what you need to consider is that the recommended values for the case num_vfs_per_pf =1 is not same as total_resources/<resource_name>.

Well, if all total_resources/<resource_name> values work in Simics we can use those as long as the tests pass. Recommended values were used just because at the point of writing that init script I didn't use total_resources/<resource_name>.

Reply:
I didn't find your recent comments regarding this, so will answer here.

> I told this does not work. Please remember that we 'used to have' what you recommended and I removed.

I remember what you've told me. I'm just not fully sure we have all required info yet for this kind of conclusion.

Please, provide the following info to continue:
- minimal values for all parameters that allow test pass for Simics and real hardware
- total_resources values for all parameters for Simics and real hardware
- max values for all parameters that can be configured for Simics and real hardware

This information would allow us to decide how we can proceed further.